full-stack-tests/backend/src/integration/SchemaSync.test.ts

```diff
@@ -0,0 +1,137 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { assert } from "chai";
+import { Suite } from "mocha";
+import { CloudSqlite, HubMock, IModelDb, IModelHost, SchemaSync, SnapshotDb } from "@itwin/core-backend";
+import { AzuriteTest } from "./AzuriteTest";
+import { HubWrappers, IModelTestUtils, KnownTestLocations } from "@itwin/core-backend/lib/cjs/test";
+import { Guid, OpenMode } from "@itwin/core-bentley";
+
+const storageType = "azure" as const;
+const containerProps = { baseUri: AzuriteTest.baseUri, storageType, containerId: "imodel-sync-itwin1", writeable: true };
+
+async function initializeContainer() {
+  await AzuriteTest.Sqlite.createAzContainer(containerProps);
+  const accessToken = await CloudSqlite.requestToken({ baseUri: AzuriteTest.baseUri, storageType: "azure", containerId: containerProps.containerId });
+  await SchemaSync.CloudAccess.initializeDb({ ...containerProps, accessToken });
+}
+
+describe("Schema synchronization", function (this: Suite) {
+  this.timeout(0);
+
+  before(async () => {
+    IModelHost.authorizationClient = new AzuriteTest.AuthorizationClient();
+    AzuriteTest.userToken = AzuriteTest.service.userToken.readWrite;
+
+    await initializeContainer();
+  });
+
+  after(async () => {
+    IModelHost.authorizationClient = undefined;
+  });
+
+  const synchronizeSchemas = async (iModel: IModelDb) => {
+    await SchemaSync.withLockedAccess(iModel, { openMode: OpenMode.Readonly, operationName: "schemaSync" }, async (syncAccess) => {
+      const uri = syncAccess.getUri();
+      iModel.nativeDb.schemaSyncPull(uri);
```

Comment:
is this function only useful for tests? I assume it happens automatically when you import schemas?

Reply:
Schema changesets are still necessary, even with the SchemaSyncDb.